### PR TITLE
Export interface MetaMaskEthereumProvider

### LINKED ISSUE #78: Export MetaMaskEthereumProvider
To use this library with TypesScript, we must export the MetaMaskEthereumProvider interface.

It is not being exported; the code change should be as simple as adding an export keyword on line 1.
https://github.com/MetaMask/detect-provider/blob/main/src/index.ts

### PR DESCRIPTION
**Pull request by httpJunkie**
Fixes #78

### CODE REVIEW

**Comment by RauppRafael** [comment]
would be nice to also include the `request` function

```ts

export type RequestArguments = {
    method: string
    params?: unknown[] | Record<string, unknown>
}

export interface MetaMaskEthereumProvider {
    isMetaMask?: boolean

    once(eventName: string | symbol, listener: (...args: any[]) => void): this

    on(eventName: string | symbol, listener: (...args: any[]) => void): this

    off(eventName: string | symbol, listener: (...args: any[]) => void): this

    addListener(eventName: string | symbol, listener: (...args: any[]) => void): this

    removeListener(eventName: string | symbol, listener: (...args: any[]) => void): this

    removeAllListeners(event?: string | symbol): this

    request<T>(args: RequestArguments): Promise<T>
}
```

File: src/index.ts
Comment (mcmire) [inline src/index.ts:1]: In order to fix the lint violations, I think we need to remove this line:

``` typescript
export = detectEthereumProvider;
```

and then update this line:

``` typescript
function detectEthereumProvider<T = MetaMaskEthereumProvider>({
```

to:

``` typescript
export default function detectEthereumProvider<T = MetaMaskEthereumProvider>({
```

This should keep the same behavior.

**Review by mcmire — COMMENTED** [review]
Hi @httpJunkie! Just wanted to put this back on your radar. I think this would be a useful improvement. Happy to push up changes and get this merged if you don't have time.

**Comment by mcmire** [comment]
@httpJunkie I've updated this branch to fit the changes in #100. I also made the exports explicit, which was an ask in #100.

@legobeat Would you mind giving this a review when you get a chance?

**Pull request merged**